src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/AudioPlayer/AudioPlayer.tsx

```diff
@@ -1,119 +1,52 @@
 import { useEffect, useState } from 'react';
 import { useTranslation } from 'react-i18next';
-import { Box } from '@mui/material';
 
-import { StyledTitleMedium, StyledBodyErrorText, theme, variables } from 'shared/styles';
+import { theme } from 'shared/styles';
 import { Svg } from 'shared/components/Svg';
-import { MLPlayer } from 'modules/Builder/components/MLPlayer';
+
 import { MediaType } from 'modules/Builder/components/MediaUploader';
 import {
   SharedToggleItemProps,
   ToggleItemContainer,
 } from 'modules/Builder/components/ToggleItemContainer';
 import { useCustomFormContext } from 'modules/Builder/hooks';
 
-import { AddAudio } from './AddAudio';
-import { UploadAudio } from './UploadAudio';
-import { RecordAudio } from './RecordAudio';
-import { RemoveAudioPopup } from './RemoveAudioPopup';
 import { StyledName, StyledNameWrapper } from './AudioPlayer.styles';
 import { AudioPlayerProps } from './AudioPlayer.types';
 import { getNameByUrl } from './AudioPlayer.utils';
+import { AudioPlayerContent } from './AudioPlayerContent';
 
 export const AudioPlayer = ({ name }: AudioPlayerProps) => {
   const { t } = useTranslation('app');
-  const [isUploadPopupOpened, setUploadPopupOpened] = useState(false);
-  const [isRecordPopupOpened, setRecordPopupOpened] = useState(false);
-  const [isRemoveAudioPopupOpened, setRemoveAudioPopupOpened] = useState(false);
-  const { setValue, watch, trigger, getFieldState } = useCustomFormContext();
+  const { watch, trigger } = useCustomFormContext();
 
   const urlName = `${name}.responseValues.file`;
   const url = watch(urlName);
   const [media, setMedia] = useState<MediaType | null>(
     url ? { url, name: getNameByUrl(url) } : null,
   );
 
-  const { error } = getFieldState(urlName);
-
   useEffect(() => {
     trigger(urlName);
     url && setMedia({ ...media, name: getNameByUrl(url) });
   }, [url]);
 
-  const onClearMedia = () => {
-    setMedia(null);
-    setValue(urlName, undefined);
-  };
-  const onCloseUploadPopup = () => setUploadPopupOpened(false);
-  const onCloseRecordPopup = () => setRecordPopupOpened(false);
-
-  const onCloseRemoveAudioPopup = () => setRemoveAudioPopupOpened(false);
-
-  const handleUploadAudio = (url?: string) => {
-    setValue(urlName, url ?? media?.url);
-    onCloseUploadPopup();
-  };
-  const handleUploadRecord = (url?: string) => {
-    setValue(urlName, url);
-    onCloseRecordPopup();
-  };
-
-  const handleRemoveAudio = () => {
-    onClearMedia();
-    onCloseRemoveAudioPopup();
-  };
-
-  const HeaderContent = ({ open }: SharedToggleItemProps) =>
+  const HeaderContent = ({ open, media }: SharedToggleItemProps & { media: MediaType | null }) =>
```

Comment:
maybe also move to it's own component?

Reply:
right, it's better to use react reconciliation independently of parent function render

Reply:
done